plugin/evm/atomic/sync/atomic_sync_summary.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package sync
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/coreth/plugin/evm/message"
+
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/crypto"
+)
+
+var _ message.Syncable = (*AtomicSyncSummary)(nil)
+
+// Codec is the codec manager that contains the codec for AtomicSyncSummary and
+// other message types that are used in the network protocol. This is to ensure that the codec
+// version is consistent across all message types and includes the codec for AtomicSyncSummary.
+var Codec codec.Manager
+
+func init() {
+	var err error
+	Codec, err = message.NewCodec(AtomicSyncSummary{})
+	if err != nil {
+		panic(fmt.Errorf("failed to create codec manager: %w", err))
+	}
+}
+
+// AtomicSyncSummary provides the information necessary to sync a node starting
+// at the given block.
+type AtomicSyncSummary struct {
+	*message.BlockSyncSummary `serialize:"true"`
+	AtomicRoot                common.Hash `serialize:"true"`
+
+	summaryID  ids.ID
+	bytes      []byte
+	acceptImpl message.AcceptImplFn
+}
+
+func NewAtomicSyncSummary(blockHash common.Hash, blockNumber uint64, blockRoot common.Hash, atomicRoot common.Hash) (*AtomicSyncSummary, error) {
```

Comment:
I understand that we only generate summaries this way when we are serving them to peers... But it feels unusual to leave `acceptImpl` undefined here and for it to only be set out-of-band in `Parse`.

Perhaps we should move the definition of `Parse` to live here as well?

Reply:
Parse is defined for `SummaryParser` which is a different type. Added comment